src/material-experimental/mdc-form-field/form-field.ts

```diff
@@ -172,10 +172,14 @@ export class MatFormField implements AfterViewInit, OnDestroy, AfterContentCheck
   set appearance(value: MatFormFieldAppearance) {
     const oldValue = this._appearance;
     this._appearance = value || (this._defaults && this._defaults.appearance) || DEFAULT_APPEARANCE;
-    // If the appearance has been switched to `outline`, the label offset needs to be updated.
-    // The update can happen once the view has been re-checked, but not immediately because
-    // the view has not been updated and the notched-outline floating label is not present.
     if (this._appearance === 'outline' && this._appearance !== oldValue) {
+      // After the view changes the appearance to outline, the label's width should be measured
+      // so that it can be provided to the notched outline.
+      this._needsOutlineNotchWidthRefresh = true;
```

Comment:
This worked before: https://github.com/angular/components/commit/e3413baf5c8380e6e086622b19fd98123506e217, but we just regressed due those changes. It surprised me initially that those changes had an effect here, but since we had the asterisk rendered through an `ngIf` template, the CDK content observer triggered and updated the notch width. This is no longer the case as of https://github.com/angular/components/commit/e3413baf5c8380e6e086622b19fd98123506e217, so the notched outline no longer works. 

I think the right fix would be to just explicitly call `_refreshOutlineNotchWidth` here, and fixing that the label scales to full width unnecessarily (causing incorrect measurement in filled). We stretch it unnecessarily due to our CSS overrides where we set `left: 0` and `right: 0`. We can set those to `initial`.

Reply:
That works much better - glad to know that `initial` is valid so we can measure the right width immediately

Reply:
Switched directly to using `left` and `right`'s default value, `auto`, since `initial` is blocklisted due to IE